va/wayland/va_wayland_linux_dmabuf.c

```diff
@@ -0,0 +1,364 @@
+    VADriverContextP const ctx = pDisplayContext->pDriverContext;
+    struct drm_state * const drm_state = ctx->drm_state;
+
+    assert(device_array->size == sizeof(dev_id));
+    memcpy(&dev_id, device_array->data, sizeof(dev_id));
+
+    dev = get_drm_device_from_dev_id(dev_id);
+    if (!dev) {
+        va_wayland_error("failed to get DRM device from device ID");
+        return;
+    }
+
+    if (!(dev->available_nodes & (1 << DRM_NODE_RENDER)))
+        goto end;
+
+    dev_path = dev->nodes[DRM_NODE_RENDER];
+    drm_state->fd = open(dev_path, O_RDWR | O_CLOEXEC);
+    if (drm_state->fd < 0) {
+        va_wayland_error("failed to open %s", dev_path);
+        goto end;
+    }
+
+    drm_state->auth_type = VA_DRM_AUTH_CUSTOM;
+
+end:
+    drmFreeDevice(&dev);
+}
+
+static void
+feedback_handle_tranche_done(
+    void                                *data,
+    struct zwp_linux_dmabuf_feedback_v1 *feedback
+)
+{
+}
+
+static void
+feedback_handle_tranche_target_device(
+    void                                *data,
+    struct zwp_linux_dmabuf_feedback_v1 *feedback,
+    struct wl_array                     *device_array
+)
+{
+}
+
+static void
+feedback_handle_tranche_formats(
+    void                                *data,
+    struct zwp_linux_dmabuf_feedback_v1 *feedback,
+    struct wl_array                     *indices_array
+)
+{
+}
+
+static void
+feedback_handle_tranche_flags(
+    void                                *data,
+    struct zwp_linux_dmabuf_feedback_v1 *feedback,
+    uint32_t                             flags
+)
+{
+}
+
+static const struct zwp_linux_dmabuf_feedback_v1_listener feedback_listener = {
+    .done = feedback_handle_done,
+    .format_table = feedback_handle_format_table,
+    .main_device = feedback_handle_main_device,
+    .tranche_done = feedback_handle_tranche_done,
+    .tranche_target_device = feedback_handle_tranche_target_device,
+    .tranche_formats = feedback_handle_tranche_formats,
+    .tranche_flags = feedback_handle_tranche_flags,
+};
+
+static void
+registry_handle_global(
+    void               *data,
+    struct wl_registry *registry,
+    uint32_t            name,
+    const char         *interface,
+    uint32_t            version
+)
+{
+    VADisplayContextP const pDisplayContext = data;
+    struct va_wayland_linux_dmabuf_context *wl_linux_dmabuf_ctx = pDisplayContext->opaque;
+    struct zwp_linux_dmabuf_v1 *linux_dmabuf;
+    struct zwp_linux_dmabuf_feedback_v1 *feedback;
+
+    if (strcmp(interface, zwp_linux_dmabuf_v1_interface.name) == 0 &&
+        version >= 4) {
+        wl_linux_dmabuf_ctx->has_linux_dmabuf = true;
+        linux_dmabuf =
+            wl_registry_bind(registry, name, &zwp_linux_dmabuf_v1_interface, 4);
+        feedback = zwp_linux_dmabuf_v1_get_default_feedback(linux_dmabuf);
+        zwp_linux_dmabuf_feedback_v1_add_listener(feedback, &feedback_listener, data);
+        zwp_linux_dmabuf_v1_destroy(linux_dmabuf);
```

Comment:
destroy here? or in handle_global_remove

Reply:
`handle_global_remove` is irrelevant here, the linux-dmabuf global is never removed by the compositor.

We destroy the `zwp_linux_dmabuf_v1` as soon as we don't need it anymore (ie, as soon as we've created a feedback object).